src/cider/nrepl/middleware/inspect.clj

```diff
@@ -165,12 +175,13 @@
       "inspect-next-sibling" next-sibling-reply
       "inspect-previous-sibling" previous-sibling-reply
       "inspect-refresh" refresh-reply
-      "inspect-get-path" get-path-reply
       "inspect-next-page" next-page-reply
       "inspect-prev-page" prev-page-reply
       "inspect-set-page-size" set-page-size-reply
       "inspect-set-max-atom-length" set-max-atom-length-reply
+      "inspect-set-max-value-length" set-max-value-length-reply
```

Comment:
While we can keep this, it would be also much welcome to have an options map so that users can simply have a defcustom.

This way we have fewer calls / moving pieces.

Existing example:

https://github.com/clojure-emacs/cider/blob/1cd6ab7b7cb4a7e079a273600c4d28307c3aba40/cider-client.el#L315-L322

Reply:
I agree that feels excessive to be able to customize all of those on the fly. Being able to set it up in one place would be better. At the same time, I don't know if a single set of options will fit everything. For example, it feels like the debugger can use stricter limitations than inspector (because debugger shows values inline with the code and inspector has a dedicated buffer).

I also think that different options have different utility in being configurable. E.g., customizing `page-size` on the fly is awesome. Being able to change `coll-length` and `coll-depth` on the fly in the inspector can be useful – if you need to see just a couple extra items or levels. But changing `atom-length` and `total-length` is of questionable benefit to me.

Reply:
Yeah I don't oppose this op, and similarly, not everything has to be exposed as an option.

Simply trying to make sure that we can get the most useful options easily conveyed from cider.el to the Inspector.

Reply:
Here's what I would do:

1. Expose all five of these parameters as defcustoms in CIDER. Let the user customize these defaults as they wish.
2. Keep page-size and coll-length dynamic ops as is. Add a new coll-depth op, bind it to `C` (shift-c) in CIDER inspector to not hijack another letter for a pretty useless command (`c` is currently for changing coll-length).
3. Drop atom-length op from cider-nrepl. Make atom-length and value-length only changeable through defcustoms.

Thoughts?

Reply:
I'd do 1 and 2. 3, I'd try really hard to avoid breaking changes.

Perhaps https://github.com/clojure-emacs/cider/discussions/2960 would be a good read - starting from that discussion (and increasingly since) we've been trying to commit to avoiding breakage.